### DISCUSSION TITLE
Which kernel should be used?

### DISCUSSION THREAD

**Discussion by wowodejiajia** [idaholab/moose#22459] [Q&A General] [answered]
Dear MOOSE experts,

![123](https://user-images.githubusercontent.com/85658962/197110624-27703266-7b47-4905-b1ae-100b9d417211.png)

L(u,a,b) and F(u) are defined in the [Materials] respectively. u is a variable. Which kernel should be used?

Thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello

~~If F is a Material then we dont define its time derivative automatically.
I think you will want to rewrite F (dF/dt really) within the kernel. This will avoid having to deal with interactions between the time integrator and a time_derivative_material_property.~~
EDIT: I saw dt not du

You can load L using getMaterialProperty in your custom kernel

Guillaume

**Reply by wowodejiajia** [reply]
If I use AllenCahn [kernel]，I think the form is the same. Can I do this
```
[kern_1]
  type = AllenCahn
  variable = u
  f_name = F
  mob_name = L
  args = 'a b'
[]
```
Thank you.

**Reply by GiudGiud** [reply]
the form looks the same indeed. I edited the syntax directly in your post

**Reply by wowodejiajia** [reply]
OK, I see. Thank you.
